Add installation guide for ESPHome with uv (#5481)
OttoWinter/esphomedocs: content/guides/installing_esphome.md

Author: @swoboda1337 Do we need `wheel`?

I've upgraded a couple of ESPhome devices in uv without having `wheel` installed

Author: We can add additional deps,

`uv tool install esphome --with wheel`

Author: After some more testing, lets add wheel and pip

uv tool install esphome --with wheel,pip

Author: I needed `wheel` during my installs, might be best to keep it to cover all cases

title: "Installing ESPHome Manually"
---

> ## Cross platform using uv

Author: Can you add the hypen?

> ## Cross platform using uv

Author: ```suggestion
## Cross-platform using uv
```

> [uv](https://docs.astral.sh/uv/) is an extremely fast Python packages and project manager, written in Rust.

Author: ```suggestion
[uv](https://docs.astral.sh/uv/) is an extremely fast Python package and project manager, written in Rust.
```

It offers a simple way to install and run ESPHome on Windows, Mac and Linux.

Once you've [installed uv](https://docs.astral.sh/uv/guides/install-uv/), you can easily [install Python](https://docs.astral.sh/uv/guides/install-python/) and Python packages.

Using uv, you can either run esphome without installing it (`uvx esphome version`) or you can install it as a [tool](https://docs.astral.sh/uv/guides/tools/) with the following command:

```shell
uv tool install esphome

> # upgrade with: uv tool upgrade esphome

Reviewer: Can probably make separate shell section for how to upgrade, looks sort of odd as comment, otherwise looks good.

Author: Thanks, I'll add a line to the end

```

Now *esphome* will be available on your PATH

```shell
esphome version
```

> 

Author: ```suggestion

Upgrade to the latest version using `uv tool upgrade esphome`
```

title: "Installing ESPHome Manually"
---

## Cross-platform using uv

[uv](https://docs.astral.sh/uv/) is an extremely fast Python package and project manager, written in Rust.

It offers a simple way to install and run ESPHome on Windows, Mac and Linux.

> Once you've [installed uv](https://docs.astral.sh/uv/guides/install-uv/), you can easily [install Python](https://docs.astral.sh/uv/guides/install-python/) and Python packages.

Reviewer: I tried this link and you get a 404: https://docs.astral.sh/uv/guides/install-uv/

Author: They moved it to https://docs.astral.sh/uv/getting-started/installation/ recently

Maybe we can exclude these links, as we already pointed them to the main uv site in the beginning?
There docs are quite clear

Reviewer: Yeah I guess, if you scroll down its right there.

> Once you've [installed uv](https://docs.astral.sh/uv/guides/install-uv/), you can easily [install Python](https://docs.astral.sh/uv/guides/install-python/) and Python packages.

Author: ```suggestion
If you don't have Python installed, uv offers an easy way to [install and manage Python versions](https://docs.astral.sh/uv/guides/install-python/).
```

Using uv, you can either run esphome without installing it (`uvx esphome version`) or you can install it as a [tool](https://docs.astral.sh/uv/guides/tools/) with the following command:

```shell

> uv tool install esphome

Author: ```suggestion
uv tool install esphome --with wheel,pip
```

# upgrade with: uv tool upgrade esphome
```

Now *esphome* will be available on your PATH

```shell
esphome version
```

> Upgrade to the latest version using `uv tool upgrade esphome`

Author: ```suggestion
Upgrade to the latest version using `uv tool upgrade esphome`

```

title: "Installing ESPHome Manually"
---

## Cross-platform using uv

[uv](https://docs.astral.sh/uv/) is an extremely fast Python package and project manager, written in Rust.

It offers a simple way to install and run ESPHome on Windows, Mac and Linux.

Once you've [installed uv](https://docs.astral.sh/uv/guides/install-uv/), you can easily [install Python](https://docs.astral.sh/uv/guides/install-python/) and Python packages.

Using uv, you can either run esphome without installing it (`uvx esphome version`) or you can install it as a [tool](https://docs.astral.sh/uv/guides/tools/) with the following command:

```shell
uv tool install esphome

> # upgrade with: uv tool upgrade esphome

Author: ```suggestion
```

> **Python 3.14 is not yet supported.** Please use Python 3.11, 3.12, or 3.13.
> Python 3.14 introduced breaking changes that ESPHome's dependencies have not yet adapted to.

## Cross-platform using uv

[uv](https://docs.astral.sh/uv/) is an extremely fast Python package and project manager, written in Rust.

It offers a simple way to install and run ESPHome on Windows, Mac and Linux.

If you don't have Python installed, uv offers an easy way to [install and manage Python versions](https://docs.astral.sh/uv/guides/install-python/).

> Using uv, you can either run esphome without installing it (`uvx esphome version`) or you can install it as a [tool](https://docs.astral.sh/uv/guides/tools/) with the following command:

Reviewer 2: maybe add also the version of esphome? for example: `uvx esphome@latest version`. as the `version` at the end could confuse a bit if you need `uvx esphome 2026.1.0` for example to install. or use another command than `version`

> Using uv, you can either run esphome without installing it (`uvx esphome version`) or you can install it as a [tool](https://docs.astral.sh/uv/guides/tools/) with the following command:

Author: ```suggestion
Using uv, you can install esphome as a [tool](https://docs.astral.sh/uv/guides/tools/) with the following command:
```
